File: backend/schemas/prescription.py
from pydantic import BaseModel, ConfigDict
from datetime import date, datetime


class MedicationItem(BaseModel):
    """Single medication item"""
    name: str
    dosage: str
    frequency: str
    duration: str


class PrescriptionCreate(BaseModel):
    """Schema for creating a prescription"""
    appointment_id: str
    patient_id: str
    diagnosis: str
    medications: list[MedicationItem]
    instructions: str | None = None
    follow_up_date: date | None = None


class PrescriptionUpdate(BaseModel):
    """Schema for updating a prescription"""
    diagnosis: str | None = None
    medications: list[MedicationItem] | None = None
    instructions: str | None = None
    follow_up_date: date | None = None


class PrescriptionResponse(BaseModel):
    """Schema for prescription response"""
    model_config = ConfigDict(from_attributes=True)

    id: str
    appointment_id: str
    patient_id: str
    doctor_id: str
    diagnosis: str
    medications: str  # JSON string
    instructions: str | None
    follow_up_date: date | None
    created_at: datetime


class PrescriptionWithDetails(PrescriptionResponse):
    """Prescription with patient and doctor names"""
    patient_name: str | None = None
    doctor_name: str | None = None

File: frontend/src/api/prescription.ts
import axios from './axios'

export interface MedicationItem {
  name: string
  dosage: string
  frequency: string
  duration: string
}

export interface PrescriptionCreate {
  appointment_id: string
  patient_id: string
  diagnosis: string
  medications: MedicationItem[]
  instructions?: string
  follow_up_date?: string
}

export interface Prescription {
  id: string
  appointment_id: string
  patient_id: string
  doctor_id: string
  diagnosis: string
  medications: string // JSON string
  instructions?: string
  follow_up_date?: string
  created_at: string // ISO datetime string
}

export const prescriptionAPI = {
  create: async (data: PrescriptionCreate): Promise<Prescription> => {
    const response = await axios.post('/prescriptions', data)
    return response.data
  },

  getMyPrescriptions: async (): Promise<Prescription[]> => {
    const response = await axios.get('/prescriptions/my-prescriptions')
    return response.data
  },

  getPatientMyPrescriptions: async (): Promise<Prescription[]> => {
    const response = await axios.get('/prescriptions/patient/my-prescriptions')
    return response.data
  },

  getPatientPrescriptions: async (patientId: string): Promise<Prescription[]> => {
    const response = await axios.get(`/prescriptions/patient/${patientId}`)
    return response.data
  },

  update: async (id: string, data: Partial<PrescriptionCreate>): Promise<Prescription> => {
    const response = await axios.put(`/prescriptions/${id}`, data)
    return response.data
  },

  delete: async (id: string): Promise<void> => {
    await axios.delete(`/prescriptions/${id}`)
  },
}
